packages/libs/eda/src/lib/core/components/computations/plugins/correlationAssayMetadata.tsx

```diff
@@ -155,3 +161,66 @@ export function CorrelationAssayMetadataConfiguration(
     </ComputationStepContainer>
   );
 }
+
+const ASSAY_ENTITIES = [
+  'OBI_0002623',
+  'EUPATH_0000809',
+  'EUPATH_0000813',
+  'EUPATH_0000812',
+];
+
+// The correlation assay x metadata app is only available for studies
+// with appropriate metadata. Specifically, the study
+// must have at least one continuous metadata variable that is on a one-to-one path
+// from the assay entity.
+// See PR #74 in service-eda-compute for the matching logic on the backend.
+function isEnabledInPicker({
+  studyMetadata,
+}: IsEnabledInPickerParams): boolean {
+  if (!studyMetadata) return false;
+
+  const entities = entityTreeToArray(studyMetadata.rootEntity);
+  // Ensure there are collections in this study. Otherwise, disable app
+  const studyHasCollections = !!entities.filter(
+    (e): e is StudyEntity & Required<Pick<StudyEntity, 'collections'>> =>
+      !!e.collections?.length
+  ).length;
+  if (!studyHasCollections) return false;
+
+  // Check for appropriate metadata
+  // Step 1. Find the first assay node. Doesn't need to be any assay in particular just any mbio assay will do
+  const firstAssayEntityIndex = entities.findIndex((entity) =>
+    ASSAY_ENTITIES.includes(entity.id)
+  );
+  if (firstAssayEntityIndex === -1) return false;
+
+  // Step 2. Find all ancestor entites of the assayEntity that are on a one-to-one path with assayEntity.
+  // Step 2a. Grab ancestor entities.
+  const ancestorEntities = ancestorEntitiesForEntityId(
+    entities[firstAssayEntityIndex].id,
+    entities
+  ).reverse(); // Reverse so that the ancestorEntities[0] is the assay and higher indices are further up the tree.
+
+  // Step 2b. Trim the ancestorEntities so that we only keep those that are on
+  // a 1:1 path. Once we find an ancestor that is many to one with its parent, we
+  // know we've hit the end of the 1:1 path.
+  const lastOneToOneAncestorIndex = ancestorEntities.findIndex(
+    (entity) => entity.isManyToOneWithParent
+  );
+  const oneToOneAncestors = ancestorEntities.slice(
+    1, // removing the assay itself
```

Comment:
So we assume there are no continuous 'metadata variables' on the *same* entity as the assay entity?

Reply:
Right. That assay entity should only have assay data.